src/vm/siginfo.cpp

```diff
@@ -4953,11 +4953,54 @@ void PromoteCarefully(promote_func   fn,
     (*fn) (ppObj, sc, flags);
 }
 
+void ReportByRefPointersFromByRefLikeObject(promote_func *fn, ScanContext *sc, PTR_MethodTable pMT, PTR_VOID pSrc)
+{
+    WRAPPER_NO_CONTRACT;
+
+    _ASSERTE(pMT->IsByRefLike());
+
+    if (pMT == g_TypedReferenceMT)
```

Comment:
It looks to me like `CEEInfo::getClassGClayout` would only return early if the topmost object is a `TypedReference`, which seems acceptable. If a by-ref-like object contains a `TypedReference` as in your example, it would call `ComputeGCLayout`, which would iterate over all fields in the by-ref-like type. I'm not seeing an issue there either.